### DISCUSSION TITLE
New Captain Perk / Ship Feature

### DISCUSSION THREAD

**Discussion by bookslayer10** [FakeFishGames/Barotrauma#11579] [Ideas and suggestions]
I have a suggestion to make playing Captain more interesting for the class, and also speed up the mission for the crew. 

I often find in ship fights that as soon as there is major flooding, you lose all maneuverability as Captain. All you can do is call out sonar readings as you sink like a rock, helpless to dodge enemies or keep yourself from skinking into the abyss. You Helm skill doesn't help with keeping the sub up because even when the ballast tanks are fully drained, hull breaches will drag you down.

Additionally, moving up and down while traveling can take a while and be tedious in some situations. You can provide bonuses to engine power to make horizontal speed greater, but you can't drain the ballast tanks past completely empty.

To improve the game, it would be cool to let the submarine engine use its power to move the sub vertically as well as horizontally.

Now, to get this feature, you would need to take a Captain perk that unlocks it, because it's the Captain that it really benefits and is used by. The perk would add a new trigger to the engine's wiring panel that activates to switch the thrust direction of the engine from horizontal to vertical.  It would need to be wired because there aren't any other good methods of precise activation for the Captain, and it could be further improved with advanced logic to switch between the modes automatically.

Because the engine usually gets the X data cable, you would either need to work with the engine interpreting forwards as up, or make it switch the cables automatically using wiring. There is a lot of interesting ship customization options here.

Not only would this add more strategy when the sub starts sinking, it would speed up dull sections of mission when you need to go up a long way.

**Comment by duarm** [comment]
Very cool, some mods solve this by adding vertical thrust, I think this would be better as an engine upgrade (maybe another engine entirely?) instead of a perk tho.

**Comment by Saevon** [comment]
When the sub is sinking with major flooding the engine would need A TON of speed to be able to counter the sink rate, or really affect any kind of dodging either.

If the engine has such power it would simply be better then the ballasts (way less space).


Adding the possibility of faster upwards speed tho would be quite nice, Tho personally I'd rather improve on the oxygen mechanics to do so!
- Add vacuum, which would be like negative oxygen.
- Rooms with vacuum would allow you to create a "ballast" which pushes you upwards faster
- This might require pump improvements which can create better and better vacuums
- Oxygen would actually be drained out of a hull which is flooded (tho only to 50% by default) and a vent would be needed to replenish it
- Vacuum would have explosive push when opened, similar to water
- Accessing improved ballast could be even more dangerous! requiring more mechanics.
- Pierced ballast becomes even worse, higher priority. As going up faster is hard.
- Pierced ballast can be vital now, as with "vacuum ballast" it could act as TWICE (or whatever upgrade/balance rating) the hull it holds for buoyancy.


Or other such systems that don't just make engine stronger
(Note massive vacuum benefits isn't really how that would work IRL, but it is Europa)
